crates/apollo-router/src/router_factory.rs

```diff
@@ -19,23 +19,41 @@ where
         &self,
         configuration: &Configuration,
         schema: Arc<graphql::Schema>,
+        query_cache_limit: usize,
     ) -> future::BoxFuture<'static, Router>;
+    fn recreate(
+        &self,
+        graph: Arc<Router>,
+        configuration: &Configuration,
+        schema: Arc<graphql::Schema>,
+        query_cache_limit: usize,
+    ) -> future::BoxFuture<'static, Router>;
+    fn get_query_cache_limit(&self) -> usize;
```

Comment:
Clearly moving `query_cache_limit` to `Configuration` will reduce the number of arguments here.

Reply:
Resolving as part of #188